modin/pandas/test/test_groupby.py

```diff
@@ -1731,6 +1785,168 @@ def test_not_str_by(by, as_index):
     eval_general(md_grp, pd_grp, lambda grp: grp.first())
 
 
+@pytest.mark.parametrize("internal_by_length", [0, 1, 2])
+@pytest.mark.parametrize("external_by_length", [0, 1, 2])
+@pytest.mark.parametrize("has_categorical_by", [True, False])
+@pytest.mark.parametrize(
+    "agg_func",
+    [
+        pytest.param(
+            lambda grp: grp.apply(lambda df: df.dtypes), id="modin_dtypes_impl"
+        ),
+        pytest.param(lambda grp: grp.apply(lambda df: df.sum()), id="apply_sum"),
+        pytest.param(lambda grp: grp.count(), id="count"),
+        pytest.param(lambda grp: grp.nunique(), id="nunique"),
+        # Integer key means the index of the column to replace it with.
+        # 0 and -1 are considered to be the indices of the columns to group on.
+        pytest.param({1: "sum", 2: "nunique"}, id="dict_agg_no_intersection_with_by"),
+        pytest.param(
+            {0: "mean", 1: "sum", 2: "nunique"},
+            id="dict_agg_has_intersection_with_by",
+        ),
+        pytest.param(
+            {1: "sum", 2: "nunique", -1: "nunique"},
+            id="dict_agg_has_intersection_with_categorical_by",
+        ),
+    ],
+)
+# There are two versions of the `handle_as_index` method: the one accepting pandas.DataFrame from
+# the execution kernel and backend agnostic. This parameter indicates which one implementation to use.
```

Comment:
@modin-project/modin-core, @modin-project/modin-omnisci, let's not use `backend` as a standalone word. It is a vague term. We have more clear words now: storage format, engine and execution.